src/compiler/checker.ts

```diff
@@ -8965,10 +8966,13 @@ namespace ts {
             if (left.kind === SyntaxKind.SuperKeyword) {
                 return true;
             }
-            // A protected property is accessible in the declaring class and classes derived from it
+            // A protected property is accessible if the property is within the declaring class or classes derived from it
+            const typeClassDeclaration = <ClassLikeDeclaration>getClassLikeDeclarationOfSymbol(type.symbol);
             if (!enclosingClass || !hasBaseType(enclosingClass, declaringClass)) {
-                error(node, Diagnostics.Property_0_is_protected_and_only_accessible_within_class_1_and_its_subclasses, symbolToString(prop), typeToString(declaringClass));
-                return false;
+                if (!isNodeWithinClass(node, typeClassDeclaration)) {

```

Comment:
that should not change. it should be 1. accessed in a protected class, and 2. through an instance of the protected class. the change is only about the first condition, not the second.